### PR TITLE
feat: sdk audit and maintenance

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

> Pull original repo changes
- Added existing features
- Added GitHub actions

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by bardisg** [comment]
@MoumitaM can we please add a changelog file with the expected list for this release?

File: .github/workflows/build-and-quality-checks.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          npm ci
+
+      - name: Execute quality checks
+        run: |
+          npm run check:circular
+          npm run check:duplicates
+
+      - name: Execute security checks
+        run: |
+          npm run check:security
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .github/workflows/build-and-quality-checks.yml:36]: Addressed

File: .github/workflows/check_pr_title.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Check PR title
+
+on:
+  pull_request:
+    branches: ['production', 'production-staging']
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .github/workflows/check_pr_title.yml]: Addressed

File: .github/workflows/build-and-quality-checks.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .github/workflows/build-and-quality-checks.yml:36]: Addressed

File: .gitignore [hunk 3]
```diff
@@ -3,5 +3,11 @@ coverage
 .nyc_output
 coverage.lcov
 lib
-./example/node_modules
-./example/package-lock.json
+examples/async-flush/node_modules
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .gitignore]: Addressed

File: .husky/commit-msg [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .husky/commit-msg:1]: Addressed

File: .husky/pre-commit [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .husky/pre-commit:1]: Addressed

File: examples/sample-app/package.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "app.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@rudderstack/rudder-sdk-node": "2.0.0"
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply examples/sample-app/package.json]: Addressed

File: index.d.ts [hunk 7]
```diff
@@ -47,22 +55,24 @@ export interface constructorOptions {
...
    * optional dictionary of `options`.
    *
    * @param {String} writeKey
-   * @param {String} dataPlaneURL
-   * @param {Object=} options (optional)
-   * @param {Number=20} options.flushAt (default: 20)
-   * @param {Number=20000} options.flushInterval (default: 20000)
-   * @param {Boolean=true} options.enable (default: true)
-   * @param {Number=20000} options.maxInternalQueueSize
+   * @param {Object} [options] (optional)
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply index.d.ts:63]: Addressed

File: sample.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const os = require('os')
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply sample.js]: Not needed so removed it.

File: index.js [hunk 9]
```diff
@@ -28,52 +29,78 @@ class Analytics {
...
-   * @param {String} dataPlaneURL
-   * @param {Object=} options (optional)
-   * @param {Number=20} options.flushAt (default: 20)
-   * @param {Number=20000} options.flushInterval (default: 20000)
-   * @param {Boolean=true} options.enable (default: true)
-   * @param {Number=20000} options.maxInternalQueueSize
+   * @param {Object} [options] (optional)
+   *   @property {Number} [flushAt] (default: 20)
+   *   @property {Number} [flushInterval] (default: 10000)
+   *   @property {String} [dataPlaneUrl] (default: 'https://hosted.rudderlabs.com')
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply index.js:37]: Addressed

File: .github/workflows/housekeeping.yaml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Run delete-old-branches-action
+        uses: beatlabs/delete-old-branches-action@v0.0.9
+        with:
+          repo_token: ${{ github.token }}
+          date: '2 months ago'
+          dry_run: false
+          delete_tags: false
+          extra_protected_branch_regex: ^(main|master|develop|production|production-staging|v1-staging|v1-production)$
```
Comment (bardisg) [inline .github/workflows/housekeeping.yaml:39]: just keep these main|master|develop

File: History.md [hunk 11]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 
+## [v2.0.0](https://github.com/rudderlabs/rudder-sdk-node/compare/v1.1.4...v2.0.0)
+
+### Features
+
+- Allow dataPlaneUrl to be be passed in options instead as an argument in initialization. [`#81`](https://github.com/rudderlabs/rudder-sdk-node/pull/81)
```
[the remark this answers is not in the document]
Comment (bardisg) [reply History.md:14]: @MoumitaM

**Review by bardisg — APPROVED** [review]

**Pull request merged**